projects/ngx-maplibre-gl/src/lib/control/geolocate-control.directive.ts

```diff
@@ -1,69 +1,70 @@
-  Host,
-  Input,
-  Output,
+  afterNextRender,
+  inject,
+  input,
+  output,
 } from '@angular/core';
-import { FitBoundsOptions, GeolocateControl } from 'maplibre-gl';
+import { type FitBoundsOptions, GeolocateControl } from 'maplibre-gl';
 import { MapService } from '../map/map.service';
 import { ControlComponent } from './control.component';
-import { Position } from '../map/map.types';
+import type { Position } from '../map/map.types';
 
 /**
  * `mglGeolocate` - a geolocate control directive
- * 
+ *
  * @category Directives
- * 
+ *
  * @see [Locate user](https://maplibre.org/ngx-maplibre-gl/demo/locate-user)
  * @see [GeolocateControl](https://maplibre.org/maplibre-gl-js/docs/API/classes/GeolocateControl)
  */
 @Directive({
   selector: '[mglGeolocate]',
   standalone: true,
 })
-export class GeolocateControlDirective implements AfterContentInit {
-  /* Init inputs */
-  @Input() positionOptions?: PositionOptions;
-  @Input() fitBoundsOptions?: FitBoundsOptions;
-  @Input() trackUserLocation?: boolean;
-  @Input() showUserLocation?: boolean;
-
-  @Output()
-  geolocate: EventEmitter<Position> = new EventEmitter<Position>();
+export class GeolocateControlDirective {
+  /* Init injection */
+  private readonly mapService = inject(MapService);
+  private readonly controlComponent = inject<
+    ControlComponent<GeolocateControl>
+  >(ControlComponent, { host: true });
 
-  constructor(
-    private mapService: MapService,
-    @Host() private controlComponent: ControlComponent<GeolocateControl>
-  ) {}
+  /* Init inputs */
+  readonly positionOptions = input<PositionOptions>();
+  readonly fitBoundsOptions = input<FitBoundsOptions>();
+  readonly trackUserLocation = input<boolean>();
+  readonly showUserLocation = input<boolean>();
 
-  ngAfterContentInit() {
-    this.mapService.mapCreated$.subscribe(() => {
-      if (this.controlComponent.control) {
-        throw new Error('Another control is already set for this control');
-      }
-      const options = {
-        positionOptions: this.positionOptions,
-        fitBoundsOptions: this.fitBoundsOptions,
-        trackUserLocation: this.trackUserLocation,
-        showUserLocation: this.showUserLocation,
-      };
+  readonly geolocate = output<Position>();
 
-      Object.keys(options).forEach((key: string) => {
-        const tkey = <keyof typeof options>key;
-        if (options[tkey] === undefined) {
-          delete options[tkey];
+  constructor() {
+    afterNextRender(() => {
+      this.mapService.mapCreated$.subscribe(() => {
+        if (this.controlComponent.control) {
+          throw new Error('Another control is already set for this control');
         }
+        const options = {
+          positionOptions: this.positionOptions(),
+          fitBoundsOptions: this.fitBoundsOptions(),
+          trackUserLocation: this.trackUserLocation(),
+          showUserLocation: this.showUserLocation(),
+        };
+
+        Object.keys(options).forEach((key: string) => {
```

Comment:
Other directives simply use if before each option, I think using the ifs is more readable, a bit verbose, but better, IMHO.

Reply:
I think keeping the "if" option is verbose and less efficient.

What we can do is create a method with an explicit name:

`const keepAvailableObjectValues = <T extends {}>(options: T): Required<T> => {
  return Object.keys(options).reduce((acc, curr) => {
    const tkey = <keyof T>curr;
    if (options[tkey] === undefined) {
      return acc;
    }
    return { ...acc, [tkey]: options[tkey] };
  }, {} as Required<T>);
};
`

And it can be use like this: 

`
      const example = keepAvailableObjectValues<ScaleControlOptions>({
        maxWidth: this.maxWidth,
        unit: this.unit,
      });
      this.controlComponent.control = new ScaleControl(example);
`

Another advantage is that it can be reused for other directives. 

What do you think ?

Reply:
This sound like a good solution, we can have this method in a util file and use it in multiple places.